shared/macros-bash.jinja

```diff
@@ -527,7 +527,11 @@ zypper install -y "{{{ package }}}"
 {{%- if pkg_manager is defined -%}}
 {{%- if pkg_manager == "yum" or pkg_manager == "dnf" -%}}
 if rpm -q --quiet "{{{ package }}}" ; then
-    {{{ pkg_manager }}} remove -y "{{{ package }}}"
+    if grep -q sudo $(find /etc/dnf/protected.d /etc/yum/protected.d -type f); then
```

Comment:
Should it be like this?
```suggestion
    if grep -q {{{ package }}} $(find /etc/dnf/protected.d /etc/yum/protected.d -type f); then
```
Not sure I agree with all `package_removed` remediations having a hardcoded check to verify if `sudo` is protected, and then removing whatever package is defined in `{{{ package }}}` (which might be protected).

Reply:
Good point, it should indeed be `{{{ package }}}` instead of `sudo` here.

Reply:
But then, the remediation becomes "remove any package, even if protected".
In this case, we might just clear the `protected_packages` list for all rules. (Not sure it is a good idea)

Reply:
Yes, it definitively should be "{{{ package }}}". Thanks for catching. Updating.
Regarding cleaning the `protected_packages` list I also agree that additional test would be interesting, like validating "if it was required by a test scenario or not", but I am not sure about the best way to do this now.
Alternatively, with less risk but similar effect is to remove using RPM directly.

Since we don't have risk considering the current rules and its respective packages, I think we can polish this conditional later, but for now simplify the test scenarios.

Reply:
> Regarding cleaning the `protected_packages` list I also agree that additional test would be interesting, like validating "if it was required by a test scenario or not", but I am not sure about the best way to do this now.

I don't think this is possible, the check for the test scenario would have to happen at build time, and the macro evaluates the `if` conditional at remediation time. It seems to me that a content that behaves differently under a testing environment would be necessary, and for that we either add conditionals used only for testing purposes (that will end up being shipped) or have a dedicated build type for testing. Neither approach looks good to me.

> Since we don't have risk considering the current rules and its respective packages, I think we can polish this conditional later, but for now simplify the test scenarios.

The conditional as it is doesn't change anything, it will always remove a package, protected or not.
Additionally, I'm not sure there is anything the remediation can check (at remediation time) to decide whether to remove a protected package or not because it is in a testing environment.

Reply:
Another alternative is to create a new macro only for testing, like `bash_package_remove_test(package)`. This, for sure, wouldn't affect remediation but would demand update in test scenario scripts. Also doable, I think.

Reply:
@marcusburghardt An analogous idea to `bash_package_remove_test` is to expand `bash_package_remove` differently for a build and a Test Suite run.
With the right Jinja conditionals this could be done. Is there a Jinja variable that is only set during an Test Suite expansion?

Reply:
I am not sure if we any common Jinja variable for Test Suite. I did some tests yesterday with this approach, treating the conditionals inside the `template.py`, but it became too complex for the value included by it, IMHO. And, probably it would be necessary to include some additional variable inside the rule.yml when calling the template. Then I moved to this much simpler approach directly on bash-macros.

Reply:
@marcusburghardt I guess a variable could be injected into `product_yaml` in https://github.com/ComplianceAsCode/content/blob/master/tests/ssg_test_suite/common.py#L429

Which can later be checked by `bash_package_remove` to expand with `protected_packages` option.